sw/legacy/demo/hello_world/main.c

```diff
@@ -99,19 +99,13 @@ int main(void) {
       // Re-enable interrupts with output complete
       arch_local_irq_enable();
 
-      // Cycling through green LEDs
-      if (USE_GPIO_SHIFT_REG) {
-        // Feed value of BTN0 into the shift register
-        set_outputs(GPIO_OUT_SHIFT, in_val);
-      } else {
-        // Cycle through LEDs unless BTN0 is pressed
-        uint32_t out_val = read_gpio(GPIO_OUT);
-        out_val          = (out_val << 1) & GPIO_LED_MASK;
-        if ((in_val & 0x1) || (out_val == 0)) {
-          out_val = 0x10;
-        }
-        set_outputs(GPIO_OUT, out_val);
+      // Cycle through LEDs
+      uint32_t out_val = read_gpio(GPIO_OUT);
+      out_val          = (out_val << 1) & GPIO_LED_MASK;
+      if (out_val == 0) {
+        out_val = 0x1;
```

Comment:
Why the functionality change here? Would it not be better to change it to 
```c
if ((in_val & (1u << 9)) || (out_val == 0)) {
```
to retain the ability to reset the LEDs by pressing up on the joystick?

Reply:
It probably was (I am unaware of the history of this demo), but in theory by reading GPIO input 1 it would have previously also worked when moving the joystick up, as the joytick navigation switches used to be on GPIO0-4 before the recent pin mapping changes. If you don't think this behaviour is relevant any more then feel free to leave it as is

Reply:
Yes, I've added this in. Thanks for the suggestion!